test/fail_compilation/b19730.d

```diff
@@ -0,0 +1,10 @@
+/*
+TEST_OUTPUT:
+---
+fail_compilation/b19730.d(9): Error: found `)` while expecting `=`
+---
+*/
+void func() {
+  bool x;
+  if (const x) {}
```

Comment:
The issue mentions `if (auto x) {}`, doesn't hurt to add a check for that, too.

Reply:
Not necessary. `const` is  like `auto`, a storage class.

Reply:
Just because its not strictly necessary doesn't mean you shouldn't add it ;)

Reply:
added